src/zjs_gfx.c

```diff
@@ -0,0 +1,468 @@
+                    74,72,72,48,58,65,65,33,122,58,66,64,32,120,0,157,160,160,
+                    125,61,66,66,66,61,61,64,64,64,61,60,36,255,36,36,72,126,73,
+                    67,102,43,47,252,47,43,255,9,41,246,32,192,136,126,9,3,32,
+                    84,84,121,65,0,0,68,125,65,48,72,72,74,50,56,64,64,34,122,
+                    0,122,10,10,114,125,13,25,49,125,38,41,41,47,40,38,41,41,41,
+                    38,48,72,77,64,32,56,8,8,8,8,8,8,8,8,56,47,16,200,172,186,
+                    47,16,40,52,250,0,0,123,0,0,8,20,42,20,34,34,20,42,20,8,85,
+                    0,85,0,85,170,85,170,85,170,255,85,255,85,255,0,0,0,255,0,
+                    16,16,16,255,0,20,20,20,255,0,16,16,255,0,255,16,16,240,16,
+                    240,20,20,20,252,0,20,20,247,0,255,0,0,255,0,255,20,20,244,
+                    4,252,20,20,23,16,31,16,16,31,16,31,20,20,20,31,0,16,16,16,
+                    240,0,0,0,0,31,16,16,16,16,31,16,16,16,16,240,16,0,0,0,255,
+                    16,16,16,16,16,16,16,16,16,255,16,0,0,0,255,20,0,0,255,0,
+                    255,0,0,31,16,23,0,0,252,4,244,20,20,23,16,23,20,20,244,4,
+                    244,0,0,255,0,247,20,20,20,20,20,20,20,247,0,247,20,20,20,
+                    23,20,16,16,31,16,31,20,20,20,244,20,16,16,240,16,240,0,0,
+                    31,16,31,0,0,0,31,20,0,0,0,252,20,0,0,240,16,240,16,16,255,
+                    16,255,20,20,20,255,20,16,16,16,31,0,0,0,0,240,16,255,255,
+                    255,255,255,240,240,240,240,240,255,255,255,0,0,0,0,0,255,
+                    255,15,15,15,15,15,56,68,68,56,68,252,74,74,74,52,126,2,2,
+                    6,6,2,126,2,126,2,99,85,73,65,99,56,68,68,60,4,64,126,32,30,
+                    32,6,2,126,2,2,153,165,231,165,153,28,42,73,42,28,76,114,1,
+                    114,76,48,74,77,77,48,48,72,120,72,48,188,98,90,70,61,62,73,
+                    73,73,0,126,1,1,1,126,42,42,42,42,42,68,68,95,68,68,64,81,
+                    74,68,64,64,68,74,81,64,0,0,255,1,3,224,128,255,0,0,8,8,107,
+                    107,8,54,18,54,36,54,6,15,9,15,6,0,0,24,24,0,0,0,16,16,0,48,
+                    64,255,1,1,0,31,1,1,30,0,25,29,23,18,0,60,60,60,60,0,0,0,0,0};
+
+u16_t maxPixels = 128 * 160 * 3;
+
+typedef struct gfx_handle {
+    jerry_value_t screenInitCB;
+    jerry_value_t drawDataCB;
+    jerry_value_t jsThis;
+} gfx_handle_t;
+
+typedef struct gfx_data {
+    u32_t coords[4];
+    u8_t color[COLORBYTES];
+    char *text;
+    jerry_size_t textSize;
+    u32_t size;
+} gfx_data_t;
+
+static jerry_value_t zjs_gfx_prototype;
+
+static void zjs_gfx_callback_free(void *native)
+{
+    // requires: handle is the native pointer we registered with
+    //             jerry_set_object_native_handle
+    //  effects: frees the gfx module
+
+    gfx_handle_t *handle = (gfx_handle_t *)native;
+    zjs_free(handle);
+}
+
+static const jerry_object_native_info_t gfx_type_info = {
+   .free_cb = zjs_gfx_callback_free
```

Comment:
I forget, but it may just work to set this to zjs_free directly here rather than adding an extra function?

Reply:
Not sure I follow, this is how we get JerryScript to call the cleanup when the object is freed.  Its the pattern used by the other modules.  Granted it will change slightly with the Jerry modules patch.

Reply:
Ah I get what you meant now, could we just execute the single line there.  As far as I know, no.  jerry_set_object_native_pointer expects a jerry_object_native_info_t for the last arg, I can't just pass it a function.